### PR TITLE
fix(naming-things): replace with_data with shallow_clone

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

> There's only one variable name worse than `data` and that's `data2`.

The method `filter.with_data` hides its true intent and thus makes the code [hard to parse](https://github.com/PostHog/posthog/pull/13830#discussion_r1086419645).

## Changes

Rename `filter.with_data` to `filter.shallow_clone`.

## How did you test this code?

I didn't 😊

### CODE REVIEW

File: posthog/api/test/__snapshots__/test_insight.ambr [hunk 1]
```diff
@@ -62,8 +62,8 @@
                        HAVING max(is_deleted) = 0) person ON person.id = pdi.person_id
                     WHERE team_id = 2
                       AND event IN ['user did things', 'user signed up']
-                      AND toDateTime(timestamp, 'UTC') >= toDateTime('2023-01-17 00:00:00', 'UTC')
-                      AND toDateTime(timestamp, 'UTC') <= toDateTime('2023-01-24 23:59:59', 'UTC')
+                      AND toDateTime(timestamp, 'UTC') >= toDateTime('2023-01-18 00:00:00', 'UTC')
```
Comment (pauldambra) [inline posthog/api/test/__snapshots__/test_insight.ambr:65]: It's surprising that this snapshot change is in this PR

**Review by pauldambra — APPROVED** [review]
rename is 💯 

but why a snapshot change?

File: posthog/api/test/__snapshots__/test_insight.ambr (hunk 1 above)
Comment (mariusandra) [reply posthog/api/test/__snapshots__/test_insight.ambr:65]: I... don't know. It... [wasn't me](https://github.com/PostHog/posthog/pull/13918/commits/aa462a0c594742b1c7cc63350b059f39129ee858) 😬

**Comment by mariusandra** [comment]
CC @macobo who seems to have added a few `with_data` calls in his last draft PR.

**Review by macobo — APPROVED** [review]
